### PR TITLE
concretize the CertDatabase interface

### PR DESCRIPTION
**Pull request by mozkeeler**
This removes some unnecessary things from FilesystemDatabase and turns it into CertDatabase. The CertDatabase interface is not necessary.

### CODE REVIEW

**Comment by smarnach** [comment]
Could you find someone from your team to do the code reviews for this? I'm happy to help with all ops-related work, but I currently don't have the bandwidth to properly review the code.

**Comment by mozkeeler** [comment]
Will do. Thanks!

**Review by jschanck — APPROVED** [review]
Looks good to me. I had to rebase on main, so I'm approving the changes now, but I'll wait to merge until after the new dev image has run.

File: go/storage/filesystemdatabase.go [hunk 1]
```diff
@@ -99,115 +91,35 @@ func (db *FilesystemDatabase) GetIssuerAndDatesFromCache() ([]IssuerDate, error)
...
-
 	serialNum := NewSerial(aCert)
 
-	certWasUnknown, err := knownCerts.WasUnknown(serialNum)
-	if err != nil {
-		return err
-	}
-
-	if certWasUnknown {
-		issuerDateSeenBefore, err := db.GetIssuerMetadata(issuer).Accumulate(aCert)
```
Comment (jschanck) [inline go/storage/filesystemdatabase.go:185]: This call to Accumulate is responsible for storing CRL and Issuer records in Redis. I've re-introduced it in the latest commit on dev.

**Comment by jschanck** [comment]
I'm going to close this since it's a PR from `dev` onto `main`. As we discussed, from now on the `dev` branch will be for experiments, and we'll open PRs to merge feature branches onto a new `landing` branch. Then we'll cut releases by merging `landing` into `main`.

**Pull request closed without merging**